apps/stripe/src/client.ts

```diff
@@ -0,0 +1,213 @@
+import Stripe from "stripe";
+
+import config from "@argos-ci/config";
+import { Account, Purchase } from "@argos-ci/database/models";
+
+import {
+  findCustomerAccountOrThrow,
+  findPlanOrThrow,
+  getClientReferenceIdPayload,
+  getEffectiveDate,
+  getFirstProductOrThrow,
+  getInvoiceCustomerOrThrow,
+  getLastPurchase,
+  getSessionCustomerIdOrThrow,
+  getSubscriptionCustomerOrThrow,
+  timestampToDate,
+  updatePurchase,
+} from "./utils.js";
+
+export type { Stripe };
+
+export const stripe = new Stripe(config.get("stripe.apiKey"), {
+  apiVersion: "2022-11-15",
+  typescript: true,
+});
+
+export const updateStripeUsage = async ({
+  account,
+  totalScreenshots,
+}: {
+  account: Account;
+  totalScreenshots: number;
+}) => {
+  try {
+    const stripeCustomerId = account.stripeCustomerId;
+    if (!stripeCustomerId) {
+      return 0;
+    }
+
+    const subscriptions = await stripe.subscriptions.list({
+      status: "active",
+      customer: stripeCustomerId,
+      expand: ["data.items"],
+    });
+    if (subscriptions.data.length > 1) {
+      throw new Error(
+        `Can't update usage. Stripe return multiple active subscriptions for account ${account.id}`
+      );
+    }
+
+    const subscription = subscriptions.data[0];
+    const meteredPricing =
+      subscription?.items.data[0]?.price.recurring?.usage_type === "metered";
+    if (!meteredPricing) {
+      return 0;
```

Comment:
`return null` instead of return 0 to distinguish the fact that you update to 0 or you didn't update.